src/main/webapp/app/course/tutorial-groups/tutorial-groups-instructor-view/tutorial-groups/crud/create-tutorial-group/create-tutorial-group.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, OnInit } from '@angular/core';
+import { AlertService } from 'app/core/util/alert.service';
+import { ActivatedRoute, Router } from '@angular/router';
+import { TutorialGroupsService } from 'app/course/tutorial-groups/services/tutorial-groups.service';
+import { TutorialGroup } from 'app/entities/tutorial-group/tutorial-group.model';
+import { onError } from 'app/shared/util/global.utils';
+import { TutorialGroupFormData } from 'app/course/tutorial-groups/tutorial-groups-instructor-view/tutorial-groups/crud/tutorial-group-form/tutorial-group-form.component';
+import { finalize } from 'rxjs/operators';
+import { HttpErrorResponse } from '@angular/common/http';
+
+@Component({
+    selector: 'jhi-create-tutorial-group',
+    templateUrl: './create-tutorial-group.component.html',
+})
+export class CreateTutorialGroupComponent implements OnInit {
+    tutorialGroupToCreate: TutorialGroup = new TutorialGroup();
+    isLoading: boolean;
+    courseId: number;
+
+    constructor(private activatedRoute: ActivatedRoute, private router: Router, private tutorialGroupService: TutorialGroupsService, private alertService: AlertService) {}
+
+    ngOnInit(): void {
+        this.activatedRoute.parent!.paramMap.subscribe((params) => {
+            this.courseId = Number(params.get('courseId'));
+        });
+        this.tutorialGroupToCreate = new TutorialGroup();
+    }
+
+    createTutorialGroup(formData: TutorialGroupFormData) {
+        // required fields
+        if (!formData?.title || !formData?.teachingAssistant) {
+            return;
+        }
+
+        const { title, teachingAssistant, additionalInformation, capacity, isOnline, language, campus } = formData;
+
+        this.tutorialGroupToCreate.title = title;
```

Comment:
using setter and returning TutorialGroup for each could also benefit you here, similar to the comment before this.

(Not writing `this.tutorialGroupToCreate` 5 times helps with not straining your arms ;))

Reply:
Could be a nice optimisation for the future but I want to get this PR done now. This does not seem critical.

Reply:
No java bean specs in typescript. Also why don't we use constructors?